pagopa/io-app
# feat: [IOCOM-871] Pay button on SEND message details, with new DS system

⚠️ This PR depends on #5661 ⚠️

## Short description
This PR adds the Pay button to a SEND's message details, with the new DS.
![Simulator Screenshot - iPhone 15 - 2024-04-04 at 10 33 26](https://github.com/pagopa/io-app/assets/5150343/ba4e6217-dec9-46b3-a25c-e6e7c58bba63)

## List of changes proposed in this pull request
- `MessageFooter` adapted to use the new DS
- `MessageFooter` added to `MessageDetails`
- Fixed bad alignment of loading "More Payments" link on `MessagePayments`

## How to test
Using the io-dev-api-server, check that:
- for a SEND message with no payments, there is no pay button
- for a SEND message with one to-pay button, there is the pay button. By clicking it, the payment flow starts and, if the payment completes, the button disappears
- for a SEND message with less or up-to five payments, all of them either paid or with an error, there is no button
- for a SEND message with less or up-to five payments, where at least one is payable, there is the pay button. By clicking it, the payments bottom sheet appears. When all of them are paid (or have an error), the button disappears
- for a SEND message, with more than five payments, no matter their statues, there is the pay button. By clicking it, the payments bottom sheet appears
- for a cancelled SEND message with no payments, there is no pay button
- for a cancelled SEND message with to-pay payments, there is no pay button
- for a cancelled SEND message with paid payments, there is no pay button
